setup.py

```diff
@@ -107,3 +109,44 @@ def run_tests(self):
         'Programming Language :: Python :: 3.6',
     ],
 )
+
+
+# The following from here to the end of the file is borrowed from wrapt's `setup.py`:
+#   https://github.com/GrahamDumpleton/wrapt/blob/4ee35415a4b0d570ee6a9b3a14a6931441aeab4b/setup.py
+# These helpers are useful for attempting build a C-extension and then retrying without it if it fails
+
+if sys.platform == 'win32':
+    build_ext_errors = (CCompilerError, DistutilsExecError, DistutilsPlatformError, IOError, OSError)
+else:
+    build_ext_errors = (CCompilerError, DistutilsExecError, DistutilsPlatformError)
+
+
+class BuildExtFailed(Exception):
+    pass
+
+
+# Attempt to build a C-extension, catch and throw a common/custom error if there are any issues
+class optional_build_ext(build_ext):
+    def run(self):
+        try:
+            build_ext.run(self)
+        except DistutilsPlatformError:
+            raise BuildExtFailed()
+
+    def build_extension(self, ext):
+        try:
+            build_ext.build_extension(self, ext)
+        except build_ext_errors:
+            raise BuildExtFailed()
+
+
+# Try to build with C extensions first, fallback to only pure-Python if building fails
+try:
+    kwargs = copy.deepcopy(setup_kwargs)
+    kwargs['ext_modules'] = [
+        Extension('ddtrace.vendor.wrapt._wrappers', ['ddtrace.vendor/wrapt/_wrappers.c']),
+    ]
+    kwargs['cmdclass']['build_ext'] = optional_build_ext
```

Comment:
This is dependent on us including `'cmdclass'` in `setup_kwargs` which currently is being done because of `tox`. However if we were to ever move away from `tox`, this would break the install. Maybe we should add a check to make sure `kwargs['cmdclass']` exists and create it if it doesn't.

I guess our CI _should_ install the module and _should_ catch this problem. But I think we can still err on the side of caution.